src/coprocessor/dag/expr/ctx.rs

```diff
@@ -47,6 +47,113 @@ pub const FLAG_DIVIDED_BY_ZERO_AS_WARNING: u64 = 1 << 8;
 pub const MODE_ERROR_FOR_DIVISION_BY_ZERO: u64 = 27;
 
 const DEFAULT_MAX_WARNING_CNT: usize = 64;
+
+pub struct EvalConfigBuilder {
+    pub tz: FixedOffset,
+    pub ignore_truncate: bool,
+    pub truncate_as_warning: bool,
+    pub overflow_as_warning: bool,
+    pub in_insert_stmt: bool,
+    pub in_update_or_delete_stmt: bool,
+    pub in_select_stmt: bool,
+    pub pad_char_to_full_length: bool,
+    pub divided_by_zero_as_warning: bool,
+    pub max_warning_cnt: usize,
+    pub sql_mode: u64,
+    pub strict_sql_mode: bool,
+}
+
+impl EvalConfigBuilder {
+    pub fn new() -> EvalConfigBuilder {
+        EvalConfigBuilder {
+            tz: FixedOffset { local_minus_utc: 8 },
+            ignore_truncate: false,
+            truncate_as_warning: false,
+            overflow_as_warning: false,
+            in_insert_stmt: false,
+            in_update_or_delete_stmt: false,
+            in_select_stmt: false,
+            pad_char_to_full_length: false,
+            divided_by_zero_as_warning: false,
+            max_warning_cnt: DEFAULT_MAX_WARNING_CNT,
+            sql_mode: 0,
+            strict_sql_mode: false,
+        }
+    }
+
+    pub fn set_timezone(&mut self, new_value: i64) -> Result<&mut EvalConfigBuilder> {
+        if tz_offset <= -ONE_DAY || tz_offset >= ONE_DAY {
+            return Err(Error::unknown_timezone(tz_offset));
+        }
+        self.tz = match FixedOffset::east_opt(tz_offset as i32) {
+            None => return Err(Error::unknown_timezone(tz_offset)),
+            Some(tz) => tz,
+        };
+        Ok(self)
+    }
+    pub fn set_ignore_truncate(&mut self, new_value: bool) -> &mut EvalConfigBuilder {
```

Comment:
I think usually it should be like `pub fn set_ignore_truncate(mut self, new_value: bool) -> Self`

Reply:
Fixed